src/routes/api/sender.js

```diff
@@ -35,6 +35,14 @@ function saveNote(req) {
         mime: 'text/html'
     });
 
+    if (req.body.label && req.body.label.trim()){
+        let value;
+        if (req.body.labelValue && req.body.labelValue.trim()){
+            value = req.body.labelValue;
+        }
+        note.setLabel(req.body.label, value);
+    }
```

Comment:
Here I would prefer to make the interface more generic and accept a list of labels.

If you're not comfortable with node I can merge this and fix to array afterwards ...

Reply:
Your decision, I'm probably done writing code for the night so you can merge and fix after, or you can commit directly to this branch so this set of changes doesn't actually go in until the android ui to support it is put together. I've turned on edits by maintainers.

I'm not sure how the UI for setting a list of key-value labels in the preferences UI will work / look in android, but I concur that your preference for the generic capability is the better choice.

Reply:
To be clear, Android app doesn't need to allow user to enter more labels at once - it can allow only once for now.

But on the server API side it's better to have this more generic since it's cheap and allow easier extension of the Android app in the future if needed.

Reply:
That sounds like a great first step- I was just testing with a hardcoded default one, so currently it allows zero :laughing: